packages/neutrino/bin/start.js

```diff
@@ -1,31 +1,50 @@
 const { start } = require('../src');
 const ora = require('ora');
+const { lookup } = require('dns');
+const { hostname } = require('os');
+const Future = require('fluture');
+
+const whenIpReady = Future.node((done) => {
+  const host = hostname();
+  lookup(host, done);
+});
 
 module.exports = (middleware, options) => {
   const spinner = ora('Building project').start();
 
-  return start(middleware, options)
-    .fork((errors) => {
-      spinner.fail('Building project failed');
-      errors.forEach(err => console.error(err));
-      process.exit(1);
-    }, (compiler) => { // eslint-disable-line consistent-return
-      if (!compiler.options.devServer) {
-        return spinner.succeed('Build completed');
-      }
+  return whenIpReady
+    .chain(ip => Future.of(process.env.HOST = process.env.HOST || ip))
```

Comment:
So now what is the behavior of `HOST` defined externally? Will server become public or private. How should it work with combination of "public" middleware setting: override or preserve? Now it will automatically become public. But I personally couldn't make it work with something different than my local IP. I can't just use any address I want. Does it work for you in v5?